src/mds/Server.cc

```diff
@@ -562,17 +583,24 @@ void Server::handle_client_session(const MClientSession::const_ref &m)
 	}
       }
 
-      session->set_client_metadata(client_metadata);
-
       if (session->is_closed())
 	mds->sessionmap.add_session(session);
 
       pv = mds->sessionmap.mark_projected(session);
       sseq = mds->sessionmap.set_state(session, Session::STATE_OPENING);
       mds->sessionmap.touch_session(session);
-      mdlog->start_submit_entry(new ESession(m->get_source_inst(), true, pv,
-					     std::move(client_metadata)),
-				new C_MDS_session_finish(this, session, sseq, true, pv));
+      auto fin = new FunctionContext([log_session_status = std::move(log_session_status)](int r){
+        if (r == 0) {
+          log_session_status("ACCEPTED", "");
+        } else {
+          CachedStackStringStream _ss;
+          auto& ss = _ss.get_stream();
+          ss << "(internal) r = " << r;
+          log_session_status("REJECTED", ss.strv());
+        }
+      });
```

Comment:
I don't think we need to log journal error. besides, it's not rejected, it's internal error

Reply:
Ya, that code is unreachable already. I'll remove it.